io.micronaut.context.exceptions.NoSuchBeanException while running on Github Actions

### Expected Behavior

Tests with test micronaut-test-resources should run on Github Actions

### Actual Behaviour

While running github actions with test resources and mongodb repositories, github actions hangs. Logs contains some errors:
```
test-resources-service] 09:58:46.617 [main] INFO  i.m.c.DefaultApplicationContext$RuntimeConfiguredEnvironment - Established active environments: [test]
[test-resources-service] 09:58:46.633 [main] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 2 test resources resolvers: io.micronaut.testresources.mongodb.MongoDBTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider
[test-resources-service] 09:58:46.679 [pool-1-thread-1] INFO  o.testcontainers.DockerClientFactory - Testcontainers version: 2.7.0
[test-resources-service] 09:58:47.059 [pool-1-thread-1] INFO  o.t.d.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
[test-resources-service] 09:58:47.084 [pool-1-thread-1] INFO  o.testcontainers.DockerClientFactory - Docker host IP address is localhost
[test-resources-service] 09:58:47.102 [pool-1-thread-1] INFO  o.testcontainers.DockerClientFactory - Connected to docker: 
  Server Version: 26.1.3
  API Version: 1.45
  Operating System: Ubuntu 22.04.5 LTS
  Total Memory: 7929 MB
[test-resources-service] 09:58:47.112 [pool-1-thread-1] INFO  o.testcontainers.images.PullPolicy - Image pull policy will be performed by: DefaultPullPolicy()
[test-resources-service] 09:58:47.114 [pool-1-thread-1] INFO  o.t.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
[test-resources-service] 09:58:47.140 [pool-1-thread-1] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling docker image: testcontainers/ryuk:0.9.0. Please be patient; this may take some time but only needs to be done once.
[test-resources-service] 09:58:48.168 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Starting to pull image
[test-resources-service] 09:58:48.183 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
[test-resources-service] 09:58:48.690 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  2 pending,  1 downloaded,  0 extracted, (8 KB/? MB)
[test-resources-service] 09:58:48.717 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  1 pending,  2 downloaded,  0 extracted, (118 KB/? MB)
[test-resources-service] 09:58:48.732 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  0 pending,  3 downloaded,  0 extracted, (118 KB/11 MB)
[test-resources-service] 09:58:48.854 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  0 pending,  3 downloaded,  1 extracted, (3 MB/11 MB)
[test-resources-service] 09:58:50.895 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  0 pending,  3 downloaded,  2 extracted, (4 MB/11 MB)
[test-resources-service] 09:58:50.950 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pulling image layers:  0 pending,  3 downloaded,  3 extracted, (11 MB/11 MB)
[test-resources-service] 09:58:50.963 [docker-java-stream--1467452668] INFO  tc.testcontainers/ryuk:0.9.0 - Pull complete. 3 layers, pulled in 2s (downloaded 11 MB at 5 MB/s)
[test-resources-service] 09:58:50.963 [pool-1-thread-1] INFO  tc.testcontainers/ryuk:0.9.0 - Image testcontainers/ryuk:0.9.0 pull took PT3.82284[41](https://github.com/Metasiteorg/demo-micronaut/actions/runs/12196727959/job/34024911875#step:5:42)54S
[test-resources-service] 09:58:50.984 [pool-1-thread-1] INFO  tc.testcontainers/ryuk:0.9.0 - Creating container for image: testcontainers/ryuk:0.9.0
[test-resources-service] 09:58:51.045 [pool-1-thread-1] INFO  tc.testcontainers/ryuk:0.9.0 - Container testcontainers/ryuk:0.9.0 is starting: 3240019e908960b6a6ce6c85b52d0ed71c2d600f91aa6382a6ee987d4b31ed25
[test-resources-service] 09:58:51.410 [pool-1-thread-1] INFO  tc.testcontainers/ryuk:0.9.0 - Container testcontainers/ryuk:0.9.0 started in PT0.[42](https://github.com/Metasiteorg/demo-micronaut/actions/runs/12196727959/job/34024911875#step:5:43)5696434S
Error: Exception in thread "main" io.micronaut.runtime.exceptions.ApplicationStartupException: Error starting Micronaut server: No bean of type [io.micronaut.context.event.ApplicationEventPublisher<io.micronaut.context.event.StartupEvent>] exists. 
	at io.micronaut.runtime.Micronaut.handleStartupException(Micronaut.java:353)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:174)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:334)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:320)
	at io.micronaut.testresources.server.TestResourcesService.main(TestResourcesService.java:[43](https://github.com/Metasiteorg/demo-micronaut/actions/runs/12196727959/job/34024911875#step:5:44))
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.context.event.ApplicationEventPublisher<io.micronaut.context.event.StartupEvent>] exists. 
	at io.micronaut.context.DefaultBeanContext.newNoSuchBeanException(DefaultBeanContext.java:2798)
	at io.micronaut.context.DefaultApplicationContext.newNoSuchBeanException(DefaultApplicationContext.java:329)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2761)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:17[45](https://github.com/Metasiteorg/demo-micronaut/actions/runs/12196727959/job/34024911875#step:5:46))
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:842)
	at io.micronaut.context.BeanLocator.getBean(BeanLocator.java:97)
	at io.micronaut.context.DefaultBeanContext.publishEvent(DefaultBeanContext.java:1831)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:360)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:216)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
	... 3 more
[test-resources-service] 09:58:[51](https://github.com/Metasiteorg/demo-micronaut/actions/runs/12196727959/job/34024911875#step:5:52).415 [pool-1-thread-1] INFO  o.t.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
[test-resources-service] 09:58:51.415 [pool-1-thread-1] INFO  o.testcontainers.DockerClientFactory - Checking the system...
[test-resources-service] 09:58:51.415 [pool-1-thread-1] INFO  o.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
> Task :internalStartTestResourcesService
```

### Steps To Reproduce

Create github actions job
```
name: Java CI with Gradle

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

jobs:
  build:

    runs-on: ubuntu-latest
    permissions:
      contents: read

    steps:
    - uses: actions/checkout@v4
    - name: Set up JDK
      uses: actions/setup-java@v4
      with:
        java-version: '21'
        distribution: 'temurin'

    - name: Setup Gradle
      uses: gradle/actions/setup-gradle@v4

    - name: Build
      run: ./gradlew build
```

### Environment Information

_No response_

### Example Application

_No response_

### Version

4.7.1

I am also facing this issue, just tried again but on micronaut platform version `4.7.3`:

```
Error: Exception in thread "main" io.micronaut.runtime.exceptions.ApplicationStartupException: Error starting Micronaut server: No bean of type [io.micronaut.context.event.ApplicationEventPublisher<io.micronaut.context.event.StartupEvent>] exists. 
	at io.micronaut.runtime.Micronaut.handleStartupException(Micronaut.java:353)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:174)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:334)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:320)
	at io.micronaut.testresources.server.TestResourcesService.main(TestResourcesService.java:43)
Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.context.event.ApplicationEventPublisher<io.micronaut.context.event.StartupEvent>] exists. 
	at io.micronaut.context.DefaultBeanContext.newNoSuchBeanException(DefaultBeanContext.java:2798)
	at io.micronaut.context.DefaultApplicationContext.newNoSuchBeanException(DefaultApplicationContext.java:329)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2761)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1745)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:842)
	at io.micronaut.context.BeanLocator.getBean(BeanLocator.java:97)
	at io.micronaut.context.DefaultBeanContext.publishEvent(DefaultBeanContext.java:1831)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:3[60]
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:216)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:75)
	... 3 more
```

The current workaround is pinning the version to `4.6.3` for running via Github actions (private org repository).

--- a/test-resources-build-tools/src/main/java/io/micronaut/testresources/buildtools/ServerUtils.java
+++ b/test-resources-build-tools/src/main/java/io/micronaut/testresources/buildtools/ServerUtils.java
@@ -81,8 +81,10 @@ public class ServerUtils {
     private static final String FLAT_JAR = "flat.jar";
 
     // See io.micronaut.testresources.testcontainers.DockerSupport.TIMEOUT
-    private static final String DOCKER_CHECK_TIMEOUT_SECONDS_ENV = "TEST_RESOURCES_DOCKER_CHECK_TIMEOUT_SECONDS";
-    private static final String DOCKER_CHECK_TIMEOUT_SECONDS_PROPERTY = "docker.check.timeout.seconds";
+    private static final String DOCKER_CHECK_TIMEOUT_SECONDS_ENV =
+        "TEST_RESOURCES_DOCKER_CHECK_TIMEOUT_SECONDS";
+    private static final String DOCKER_CHECK_TIMEOUT_SECONDS_PROPERTY =
+        "docker.check.timeout.seconds";
 
     /**
      * Writes the server settings in an output directory.
@@ -335,16 +337,24 @@ public class ServerUtils {
         waitForServerToBeAvailable(serverFactory, explicitPort, portFilePath);
     }
 
-    private static void waitForServerToBeAvailable(ServerFactory serverFactory, Integer explicitPort,
-                                  Path portFilePath) {
+    private static void waitForServerToBeAvailable(ServerFactory serverFactory,
+                                                   Integer explicitPort,
+                                                   Path portFilePath) {
         Integer port = explicitPort;
         if (explicitPort == null) {
-            while (!Files.exists(portFilePath)) {
+            int retries = 4;
+            long dur = STARTUP_TIME_WAIT_MS;
+            while (--retries > 0 && !Files.exists(portFilePath)) {
                 try {
-                    serverFactory.waitFor(Duration.of(STARTUP_TIME_WAIT_MS, ChronoUnit.MILLIS));
+                    serverFactory.waitFor(Duration.of(dur, ChronoUnit.MILLIS));
                 } catch (InterruptedException e) {
                     Thread.currentThread().interrupt();
+                    break;
                 }
+                dur *= 2;
+            }
+            if (!Files.exists(portFilePath)) {
+                throw new IllegalStateException("Port file not created. Server probably failed to start.");
             }
             try {
                 port = Integer.parseInt(Files.readString(portFilePath));

